src/OrchardCore/OrchardCore.Abstractions/Localization/ILocalizationDataProvider.cs

```diff
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace OrchardCore.Localization
+{
+    public interface ILocalizationDataProvider
+    {
+        IEnumerable<string> GetAllStrings();
```

Comment:
We'll need something different than just strings, because we'll need some "context" to translate something.

Imagine the user only sees "Article", does it mean the type, the field name in a type, a role, a menu item?
So each result should be a string but also two LocalizedString that describe what this string represents so we can group them.

For instance the `ContentFieldDataLocalizationProvider` would return an instance like this:
Context: "Content Types"
Text: "Article"

And for the fields:
Context: "Content Types - Article" (translated from T["Content Types - {0}", contentype.Name] in the provider)
Text: "Summary"

Notice that the content type is not translated in the Context value, the same way it is not translated in the po files msdgid. This is a reference text, and that's fine. It's in the same language as the source one that needs to be translated.

Reply:
Let me rethink about this and get back to you

Reply:
If you are confused, remember that this interface is only used by the screen that allows for the edition on the translations themselves. You could definitely have the recipe support and full localization support without it if you finish by the editor screen. Which is why I am surprised to see it already in this PR. Note I already made the same comment in the issue, stating that it was in point 1 when it's actually only necessary later on.

Reply:
There's no confusion at all :) just rethinking of it from many perspectives

> remember that this interface is only used by the screen that allows for the edition on the translations themselves

Ya it should be used in the admin screen by injecting `IEnumerable<ILocalizationDataProvider>` to aggregate all the localization data providers

Reply:
We'll need something different than just strings, because we'll need some "context" to translate something.

> Imagine the user only sees "Article", does it mean the type, the field name in a type, a role, a menu item?

I as a translator it doesn't matter to me if the "Article" is a content type, field or menu item, because at the end the translation is same and usually will be one or two words - unless you have an example -

> So each result should be a string but also two LocalizedString that describe what this string represents so we can group them.

I'm not against the context, but IMHO we can group them by the provider itself, "ContentField" for `ContentFieldDataLocalizationProvider`, "ContentType" for `ContentTypeDataLocalizationProvider` and so on

For instance the ContentFieldDataLocalizationProvider would return an instance like this:
Context: "Content Types"
Text: "Article"

> Context: "Content Types - Article" (translated from T["Content Types - {0}", contentype.Name] in the provider)
Text: "Summary"

This is sort of context is useful for translating a phrase like what we have in po files, also it require sort of extraction tool, but while we are talking about the dynamic data this will be tricky and insufficient, correct me if I'm wrong

To clarify what I'm talking about, let us have the following:
```json
{
    "ContentType": [            // Context
         {
             name : "Article",    // key or msgid
             translations: [
                 "en-US": "Article",
                 "fr": "Article",
                 "es": "Artículo"
             ]
         },
         ...
    ],
    "ContentField": [
         {
             name : "Name",    // key or msgid
             translations: [
                 "en-US": "Name",
                 "fr": "Nome",
                 "es": "Nombre"
             ]
         },
         ...
    ],
    ...
}
```

I as translator everything is clear to me, so I can translate and know what's the thing I'm translating: type, field .. etc

Some features that may or might not be included:
- Provide a tool to make the translation easier something like Crowding (As an open source project)
- Make the exporter extensible: json, po .. etc
- Integrate with translation provider such as: Google Translator, Bing .. etc
- Translate once reflect everyone, for instance if we have "Name" as field for many content types, we can translate it once and will reflect to all occurrences

Seb your feedback is very important to me either positive or negative

thanks

Reply:
Just a couple of thoughts @hishamco 

Context is important to me, and as with everything it Orchard Core it will always be used in ways you cannot imagine. 
So I would agree that the providers need to be able to include context, that they can figure out themselves.

I'm not a translator, I'm a dev, but I give you an example of the last set of translations I have to get done - with little or no context available (it was so much fun)

1. Export translation set
2. Give to project manager
3. Explain to project manager in detail where each string is used, and justify why we are paying for translations on them
4. Remove unwanted translations from source code
5. Return to project manager 
6. project manager sends to client
7. Project Manager explains to client where each string was used
8. Project Manage sends to translator
9. Translator contacts to understand context of translation, to chose correct...
10. Receive translations back
11. Release to stage
12. Pull from stage due to bad formatting of translations
13. Rinse and repeat. 

Total time for translations 8 months.

Reply:
As I said before I'm not against the context at all but extract them a headache it requires me frequently to extract the localization string as what's happening with po files currently

> 3. Explain to project manager in detail where each string is used, and justify why we are paying for translations on them

Simply all the extracted translations are entered by the client himself, don't forgot this a dynamic data localization

> 4. Remove unwanted translations from source code

No need for this, because again it's already entered by the client, and all these translations represents the content types, fields, menu items that the client originally used

> 7. Project Manager explains to client where each string was used

The exported translations is already grouped by context which in this case the localization data provider itself, so for validation the someone can check the content types, content fields .. etc in the OrchardCore, which is unnecessary step because he doesn't TRUST Orchard Core :)

At the end the process should be easier than we think:

1. Export the data localizations set
2. Send it to PM
3. PM forward it to the translator
and BOOM it's done

I'm agree that the case on content localization not data localization, thanks to your feedback @deanmarcussen